### DISCUSSION TITLE
Event Driven Input Handeling with Signals

### DISCUSSION THREAD

**Discussion by Mohammad9760** [godotengine/godot-proposals#9033] [Scripting]
Hi!

it'd be nice to be able to handle inputs without using conditionals.
so there could be a function that returns a signal for input actions that is emitted when the input is performed.

a simple example of how this can change the code:
```
func _process(_delta):
	if Input.is_action_just_released("ui_right"):
		roll(Vector3.RIGHT)
```

this could instead be done like this:

```
func _ready():
	Input.get_action_signal("ui_right").connect(func() : roll(Vector3.RIGHT) )
```

there could be more parameters that enables you to get some data from this signal as well and to specify what interaction you need.

but the general idea is that you no more handle inputs with conditionals, but instead connecting functions to input action signals.

I believe it's doable in GDscript using Signals, and in C# using events.

**Comment by AThousandShips** [comment]
Thank you for your suggestion!

Closing as a duplicate of:
* https://github.com/godotengine/godot-proposals/issues/1853

**Reply by Mohammad9760** [reply]
I ask to reopen this since this is not a duplicate of the other proposal.
I'm not suggesting a function that gets called automatically. I'm suggesting that you would be able to get a signal for the input actions that you add in the input map. with that it wouldn't necessarily mean that you have a function. you could have a lambda expression connected to the signal of an input action.

**Reply by AThousandShips** [reply]
It's the same, that's how a signal works, it doesn't have to be a function, see especially [this](https://github.com/godotengine/godot-proposals/issues/1853#issuecomment-1501098437) comment which suggests exactly what you suggest

So this is the same idea and feature, so shouldn't be discussed in two different places 🙂 and even if it wasn't the exact same idea it doesn't make sense to have two different suggestions that are very similar open at the same time

**Reply by Mohammad9760** [reply]
ok
